### PR TITLE
docker builds: fix cpp toolchain support

### PR DESCRIPTION
**Pull request by SchmErik**
This change fixes errors that occur from building guests that use C code (such as c-kzg) using the docker reproducible build image. This change set adds the `CC` and `CFLAGS` to the appropriate location and uses a new docker image that includes the `build-essentials` package.

### CODE REVIEW

**Comment by SchmErik** [comment]
cc @hashcashier for vis

File: risc0/build/src/docker.rs [hunk 1]
```diff
@@ -154,14 +154,19 @@ fn create_dockerfile(
...
         .copy(".", ".")
         .env(manifest_env)
         .env(rustflags_env)
         .env(&[("CARGO_TARGET_DIR", "target")])
         // Fetching separately allows docker to cache the downloads, assuming the Cargo.lock
         // doesn't change.
+        .env(&[(
+            "CC",
+            "/root/.local/share/cargo-risczero/cpp/bin/riscv32-unknown-elf-gcc",
+        )])
```
Comment (flaub) [inline risc0/build/src/docker.rs:168]: I would imagine we'd only want to configure the guest-side, this will also include the host-side.

Comment (SchmErik) [reply risc0/build/src/docker.rs]: Good point - we don't have any spaces in the gcc path so I pushed a change just now to use the `CC_riscv32im_risc0_zkvm_elf` flag to keep it confined to the guest

File: risc0/build/src/docker.rs [hunk 2]
```diff
@@ -154,14 +154,19 @@ fn create_dockerfile(
...
         .env(manifest_env)
         .env(rustflags_env)
         .env(&[("CARGO_TARGET_DIR", "target")])
         // Fetching separately allows docker to cache the downloads, assuming the Cargo.lock
         // doesn't change.
+        .env(&[(
+            "CC_riscv32im_risc0_zkvm_elf",
+            "/root/.local/share/cargo-risczero/cpp/bin/riscv32-unknown-elf-gcc",
+        )])
+        .env(&[("CFLAGS_riscv32im_risc0_zkvm_elf", "-march=rv32im -nostdlib")])
```
Comment (flaub) [inline risc0/build/src/docker.rs:169]: Could we move these lines up before the comment? I think the comment should stay near the `.run(&fetch_cmd)`

Comment (SchmErik) [reply risc0/build/src/docker.rs:167]: thanks for catching this!

**Review by flaub — APPROVED** [review]

**Pull request merged**